functionary/core/tests/api/v1/views/test_task.py

```diff
@@ -228,6 +230,36 @@ def mock_file_upload(_task, _request):
     assert Task.objects.filter(id=task_id).exists()
 
 
+def test_fail_file_upload(
+    mocker,
+    admin_client: Client,
+    file_function: Function,
+    request_headers: dict,
+):
+    def mock_file_upload(_task, _request):
+        """Mock the method of uploading a file to S3"""
+        raise S3FileUploadError("Failed to upload file")
+
+    url = reverse("task-list")
+
+    mocker.patch("core.api.v1.views.task.handle_file_parameters", mock_file_upload)
+
+    example_file = BytesIO(b"Hello World!")
+    task_input = {"function": str(file_function.id), "prop1": example_file}
+    response = admin_client.post(
+        url,
+        data=task_input,
+        content_type=MULTIPART_CONTENT,
+        **request_headers,
+    )
+
+    task_id = response.data.get("id")
+
+    assert response.status_code == status.HTTP_503_SERVICE_UNAVAILABLE
+    assert task_id is None
+    assert not Task.objects.filter(id=task_id).exists()
```

Comment:
This assertion seems superfluous given the previous one.  If `assert task_id is None` passes, this is guaranteed to be false.  In the similar test for the ui view you filter on the `file_function` which seems like a good alternative.